packages/editor-ui/src/components/TextWithHighlights.vue

```diff
@@ -0,0 +1,39 @@
+<script lang="ts" setup>
+import type { PropType } from 'vue';
+import { GenericValue } from 'n8n-workflow';
+
+const props = defineProps({
+	content: {
+		type: [Object, String, Number] as PropType<GenericValue>,
+	},
+	search: {
+		type: String,
+	},
+});
+
+const splitTextBySearch = (text = '', search = ''): { tag: 'span' | 'mark'; content: string }[] => {
+	if (!search) {
+		return [
+			{
+				tag: 'span',
+				content: text,
+			},
+		];
+	}
+
+	const pattern = new RegExp(`(${search})`, 'g');
+	const splitText = text.split(new RegExp(pattern, 'gi'));
+
+	return splitText.map((t) => ({ tag: pattern.test(t) ? 'mark' : 'span', content: t }));
+};
+</script>
+
+<template>
+	<span v-if="props.search && typeof props.content === 'string'">
```

Comment:
What happens if the content is not a string? We are providing the search box but nothing will be highlighted?

Reply:
it actually works with both boolean and number
<img width="791" alt="Screenshot 2023-12-22 at 13 48 59" src="https://github.com/n8n-io/n8n/assets/4711238/8a934046-fbf6-4c8e-92b8-3266f20610d2">